auslib/db.py

```diff
@@ -928,16 +929,20 @@ def __init__(self, db, dialect, metadata, baseTable, conditions=("time", "uptake
             # See https://github.com/zzzeek/sqlalchemy/blob/rel_0_7/lib/sqlalchemy/schema.py#L781
             # for background.
             newcol.key = newcol.name = "base_%s" % col.name
-            # 2) Primary keys on the base table are normal columns here,
-            # and are allowed to be null (because we support adding new rows
-            # to tables with autoincrementing keys via scheduled changes).
-            # The base table's data version may also be null for the same reason.
-            if col.primary_key or newcol.name == "base_data_version":
+            # 2) All non PK columns from baseTable should be nullable. Base table PK columns
+            # which are autoincrement should be nullable in the Scheduled Changes version of
+            # the column. Base table PK columns which are not autoincrement should not be nullable.
```

Comment:
Thanks for this - why aren't non-integer PK columns allowed to be null? I'm looking at some recent pull requests to find out, but it would be super helpful if you could point me in the right direction or give me an idea of what I'm looking for. Thanks again!